### ISSUE TITLE
Shall we move more things to taps?

### ISSUE THREAD

**Issue by mikemcquaid** [Homebrew/homebrew#38819]
I'd like to move anything that has a hard dependency on non-Homebrew or Cask things to taps. I'm thinking particularly of `:tex`, `:osxfuse` and hard `:x11` dependencies to separate taps each.

On the X11 dependency front (maybe a separate issue for discussion here) I'm thinking perhaps we either have a dedicated GTK/GNOME tap or start defaulting GTK packages to building without X11 support by default.

@Homebrew/owners thoughts?

**Comment by xu-cheng** [comment]
I agree to move `:tex` into a separate tap. And the same time, I wish ci-bot can install BasicTeX. Therefore, we can bottle them.

**Comment by asparagui** [comment]
+1 on getting fuse into its own taps.

**Comment by DomT4** [comment]
If someone can help me iron out using Travis as a brew test-bot (i.e. only building each PR, not every commit, every formula each time, etc), I've nearly got the ` :tex ` stuff working here: https://github.com/DomT4/homebrew-tex

> :osxfuse

:+1: :tada:.

**Comment by bcomnes** [comment]
I wouldn't mind this as long as people get hinting about the existence of these taps if they haven't tapped them yet.

**Comment by DomT4** [comment]
DomT4/Tex has become [Homebrew/Tex](https://github.com/Homebrew/homebrew-tex). Migration underway in #38855 and https://github.com/Homebrew/homebrew-tex/pull/3.

**Comment by asparagui** [comment]
^^ :+1: :facepunch:

**Comment by shrx** [comment]
@bcomnes makes a valid point, +1

**Comment by mikemcquaid** [comment]
@bcomnes They'll show up in `brew search`.

**Comment by DomT4** [comment]
@mikemcquaid How do you want to deal with `osxfuse` given we have the `osxfuse` requirement, which has a default_formula written into it. Are you happy with a default formula from another tap, or do you want to drop `osxfuse` (the formula) entirely and enforce using the upstream binary, or go down the `homebrew-php` Requirements route, or?

**Comment by mikemcquaid** [comment]
@DomT4 I think putting them into a tap perhaps with the `osxfuse` formula itself seems like a good idea. `homebrew-fuse` or something?

**Comment by DomT4** [comment]
Sounds good :+1:. Do you want to move the requirement into the tap as well, or do we just point the `default_formula` at the new tap?

**Comment by mikemcquaid** [comment]
Let's move the requirement in too, good idea. I'd put the binary `osxfuse` formula in there too.

**Comment by DomT4** [comment]
>  I'd put the binary osxfuse formula in there too.

We don't ship the upstream binary? Unless you mean the one from Cask, or I'm misunderstanding?

**Comment by mikemcquaid** [comment]
@DomT4 I think there's a formula in `homebrew-binary` and, if not, might want to create one to put in that repo.

**Comment by DomT4** [comment]
Mandatory `:tex`, Fuse and `:x11` formulae now reside in `homebrew/tex`, `homebrew/fuse` and `homebrew/x11` respectively :tada:.
